src/librustdoc/html/static/main.js

```diff
@@ -1837,14 +1851,19 @@ function defocusSearchBar() {
                 var crateSize = 0;
 
                 searchWords.push(crate);
-                searchIndex.push({
+                var crateRow = {
                     crate: crate,
                     ty: 1, // == ExternCrate
                     name: crate,
                     path: "",
                     desc: rawSearchIndex[crate].doc,
+                    parent: undefined,
```

Comment:
There is not much point into adding a field if it's to set it to `undefined`.

Reply:
There are two reasons for doing that.

One of them is to try to coax it into using a single [implicit class](xx) for both crate objects and other items. Maybe I should route them both through a constructor, just to be sure it works?

The other reason is to prevent it from walking the prototype chain when the code checks for that field. By explicitly setting it to `undefined `, it will find it and stop immediately instead.